docs/rackhd/npm_based_installation.rst

```diff
@@ -0,0 +1,242 @@
+
+   * ens192 for dhcp/pxe to boot/configure the machines
+
+* Edit the network:
+
+   * ens160 - assign IP address as appropriate for the environment, or you can use DHCP
+
+   * ens192 static ( 172.31.128.0/22 )
+
+     this is the 'default'. it can be changed, but more than one file needs to be changed.)
+
+**Packages**
+
+1. **NodeJS 4.x**
+
+   **If Node.js 4.x is not installed**
+
+* **Remove Node.js (< 4.0)**
+
+   *If Node.js is installed via apt, but is older than version 4.x, do this first* (apt-get installs v0.10 by default)
+   .. code::
+
+    sudo apt-get remove nodejs nodejs-legacy
+
+* **Install Node.js 4.x**
+
+   Add the NodeSource key and repository (*instructions copied from* https://github.com/nodesource/distributions#manual-installation):
+
+   .. code::
+
+    curl --silent https://deb.nodesource.com/gpgkey/nodesource.gpg.key | sudo apt-key add -
+    VERSION=node_4.x
+    DISTRO="$(lsb_release -s -c)"
+    echo "deb https://deb.nodesource.com/$VERSION $DISTRO main" | sudo tee /etc/apt/sources.list.d/nodesource.list
+    echo "deb-src https://deb.nodesource.com/$VERSION $DISTRO main" | sudo tee -a /etc/apt/sources.list.d/nodesource.list
+
+    sudo apt-get update
+    sudo apt-get install nodejs
+
+* **Ensure Node.js is at version 4.x**
+
+   For Example:
+
+   .. code::
+
+    $ node -v
+    v4.4.5
+
+
+2. **Dependencies**
+
+* **Install dependency pakcages**
+
+   .. code::
+
+    sudo apt-get install rabbitmq-server
+    sudo apt-get install mongodb
+    sudo apt-get install snmp
+    sudo apt-get install ipmitool
+    
+    sudo apt-get install git
+    sudo apt-get install unzip
+    sudo apt-get install ansible
+    sudo apt-get install apt-mirror
+    sudo apt-get install amtterm
+
+    sudo apt-get install isc-dhcp-server
+
+
+####
+
+Install & Configure RackHD
+^^^^
+
+1. **Install RackHD NPM package**
+
+*
+   .. code::
+
+     for service in $(echo "on-dhcp-proxy on-http on-tftp on-syslog on-taskgraph");
+     do npm install $service;
+     done
+
+
+2. **Basic RackHD Configuration**
+
+
+* **DHCP**
+
+   Update /etc/dhcp/dhcpd.conf per your network configuration
+ 
+   .. code::
+
+    # RackHD added lines
+    deny duplicates;
+
+    ignore-client-uids true;
+
+    subnet 172.31.128.0 netmask 255.255.240.0 {
+      range 172.31.128.2 172.31.143.254;
+      # Use this option to signal to the PXE client that we are doing proxy DHCP
+      option vendor-class-identifier "PXEClient";
+    }
+
+
+#######
+
+* **RACKHD APPLICATIONS**
+
+   Create the required file /opt/monorail/config.json , you can use the demonstration configuration file at https://github.com/RackHD/RackHD/blob/master/packer/ansible/roles/monorail/files/config.json as a reference.
+
+#######
+
+* **RACKHD BINARY SUPPORT FILES**
+
+   Downloaded binary files from bintray.com/rackhd/binary and placed them.
+
+   .. code::
+
+    #!/bin/bash
+
+    mkdir -p node_modules/on-tftp/static/tftp
```

Comment:
Why the folder name has a leading word `node_modules`? the on-tftp is a service, it is not put under the `node_modules`, so I assume this is a typo that need fix.

The same for below line and the on-http static files downloading.